src/app/src/pages/redteam/report/components/RiskCategories.tsx

```diff
@@ -20,57 +33,315 @@ const RiskCategories = ({
+    () => new Set(Object.keys(riskCategories)),
+  );
+  const [drawerOpen, setDrawerOpen] = useState(false);
+  const [selectedPlugin, setSelectedPlugin] = useState('');
+
+  const categories = Object.keys(riskCategories)
+    .map((category) => {
+      const categoryName = category as TopLevelCategory;
+      const subCategories = riskCategories[categoryName];
+
+      const totalPasses = subCategories.reduce(
+        (acc, subCategory) => acc + (categoryStats[subCategory]?.pass || 0),
+        0,
+      );
+      const totalTests = subCategories.reduce(
+        (acc, subCategory) => acc + (categoryStats[subCategory]?.total || 0),
+        0,
+      );
+
+      const testTypes = subCategories
+        .map((subCategory) => {
+          const stats = categoryStats[subCategory];
+          const numPassed = stats?.pass || 0;
+          const totalForSubcategory = stats?.total || 0;
+          const numFailed = totalForSubcategory - numPassed;
+
+          return {
+            name: subCategory,
+            categoryPassed: numPassed === totalForSubcategory && totalForSubcategory > 0,
+            numPassed,
+            numFailed,
+            total: totalForSubcategory,
+          };
+        })
+        .filter((test) => test.total > 0);
+
+      return {
+        name: category,
+        description: categoryDescriptions[categoryName as keyof typeof categoryDescriptions],
+        totalPasses,
+        totalTests,
+        testTypes,
+        passRate: totalTests > 0 ? totalPasses / totalTests : 1,
+      };
+    })
+    .filter((category) => category.totalTests > 0);
+
+  const toggleCategory = (categoryName: string) => {
+    setExpandedCategories((prev) => {
+      const next = new Set(prev);
+      if (next.has(categoryName)) {
+        next.delete(categoryName);
+      } else {
+        next.add(categoryName);
+      }
+      return next;
+    });
+  };
+
+  const handlePluginClick = (pluginName: string) => {
+    setSelectedPlugin(pluginName);
+    setDrawerOpen(true);
+  };
+
+  const getPassRateStyles = (passRate: number): { bg: string; text: string } => {
+    if (passRate >= 0.9) {
+      return {
+        bg: 'bg-emerald-500',
+        text: 'text-emerald-600 dark:text-emerald-400',
+      };
+    }
+    if (passRate >= 0.7) {
+      return {
+        bg: 'bg-amber-500',
+        text: 'text-amber-600 dark:text-amber-400',
+      };
+    }
+    if (passRate >= 0.5) {
+      return {
+        bg: 'bg-orange-500',
+        text: 'text-orange-600 dark:text-orange-400',
+      };
+    }
+    return {
+      bg: 'bg-red-500',
+      text: 'text-red-600 dark:text-red-400',
+    };
+  };
+
+  if (categories.length === 0) {
+    return null;
+  }
+
+  // Calculate overall stats
+  const overallPassed = categories.reduce((acc, cat) => acc + cat.totalPasses, 0);
+  const overallTotal = categories.reduce((acc, cat) => acc + cat.totalTests, 0);
+  const overallPassRate = overallTotal > 0 ? overallPassed / overallTotal : 1;
 
   return (
```

Comment:
can you break these up into smaller components? I think each risk category row could be a component